Mrc 1693 adds uncertainty ranges to the tables

Nicolas-Dolan opened the pull request:
Added columns for upper and lower uncertainty ranges for each indicator to the table. It might need further changes to the backend to work more efficiently (see comment in diff).

Nicolas-Dolan commented on src/app/static/src/app/components/plots/table/Table.vue:
```diff
@@ -7,12 +7,25 @@
+                          but I'll need Rob to pass them into indicators for the choropleth/bubbleplot from the backend first.
+                          Currently they are only available for the indicators of the barchart. I'll also need to add a v-translate  
+                          for upper/lower-->
+                          <th v-if="filteredData[0].indicatorValues[i.indicator].upper">Upper</th>
```
When I spoke with UNAIDS, I specifically asked about this. They said their preferences was having the uncertainty ranges in separate columns because it makes it easier to sort/filter that way.

Nicolas-Dolan commented:
I changed the model output getters in this file: hint/src/app/static/src/app/store/modelOutput/modelOutput.ts

This seems to have had a knock on effect on a lot of different files so it appears like I've changed a lot more files in the diff than I have. Other files to look at are the table component and its test file + the translation file, which I added translation for upper/lower
hint/src/app/static/src/app/store/translations/locales.ts

These changes will conflict with changes in the sort and filter ticket (1699) so I would prefer that 1699 was merged in first and then I can adapt this ticket to work with the new sortable table. Jeff mentioned that he would prefer the uncertainty ranges not to be on their own columns (in contrast to what UNAIDS want) so we have to decide how to present them. Perhaps in small text below the indicators in question like we did for the area hierarchies?

hillalex commented:
> I changed the model output getters in this file: hint/src/app/static/src/app/store/modelOutput/modelOutput.ts
> 
> This seems to have had a knock on effect on a lot of different files so it appears like I've changed a lot more files in the diff than I have. Other files to look at are the table component and its test file + the translation file, which I added translation for upper/lower
> hint/src/app/static/src/app/store/translations/locales.ts
> 
> These changes will conflict with changes in the sort and filter ticket (1699) so I would prefer that 1699 was merged in first and then I can adapt this ticket to work with the new sortable table. Jeff mentioned that he would prefer the uncertainty ranges not to be on their own columns (in contrast to what UNAIDS want) so we have to decide how to present them. Perhaps in small text below the indicators in question like we did for the area hierarchies?

Looks like you've got a copy of hintr-master in the app/static folder! Can you just delete that first and push that change?

Nicolas-Dolan commented:
> > I changed the model output getters in this file: hint/src/app/static/src/app/store/modelOutput/modelOutput.ts
> > This seems to have had a knock on effect on a lot of different files so it appears like I've changed a lot more files in the diff than I have. Other files to look at are the table component and its test file + the translation file, which I added translation for upper/lower
> > hint/src/app/static/src/app/store/translations/locales.ts
> > These changes will conflict with changes in the sort and filter ticket (1699) so I would prefer that 1699 was merged in first and then I can adapt this ticket to work with the new sortable table. Jeff mentioned that he would prefer the uncertainty ranges not to be on their own columns (in contrast to what UNAIDS want) so we have to decide how to present them. Perhaps in small text below the indicators in question like we did for the area hierarchies?
> 
> Looks like you've got a copy of hintr-master in the app/static folder! Can you just delete that first and push that change?

This is now done.

hillalex commented:
Thanks, there are also 5 files that need to be deleted called `master.zip` `master.zip.1` etc, I take it from failed attempts to regenerate types. You'll also have to merge the latest master into this branch to fix the test failures (since all the db tables were renamed.) When that's done can you mark it as ready for review?

EmmaLRussell commented:
I agree that it would nice to present the errors with a consistent style with the area hierarchy. Check with Jeff what a good text format would be for the values. 

>Jeff mentioned that he would prefer the uncertainty ranges not to be on their own columns (in contrast to what UNAIDS want) so we have to decide how to present them. Perhaps in small text below the indicators in question like we did for the area hierarchies?
